lib/api/LogManagerImpl.cpp

```diff
@@ -807,12 +807,53 @@ namespace MAT_NS_BEGIN
     void LogManagerImpl::SetDataInspector(const std::shared_ptr<IDataInspector>& dataInspector)
     {
         LOCKGUARD(m_dataInspectorGuard);
-        m_dataInspector = dataInspector;
+        if(dataInspector == nullptr)
+        {
+            LOG_WARN("Attempting to set nullptr as DataInspector");
+            return;
+        }
+
+        auto itDataInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&dataInspector](const std::shared_ptr<IDataInspector>& currentInspector)
+        {
+            return strcmp(typeid(*dataInspector).name(),typeid(*currentInspector).name()) == 0;
+        });
+
+        if (itDataInspector != m_dataInspectors.end())
+        {
+            LOG_WARN("Replacing specified IDataInspector with passed in inspector");
+            m_dataInspectors.erase(itDataInspector);
+        }
+
+        m_dataInspectors.push_back(dataInspector);
+    }
+
+    void LogManagerImpl::ClearDataInspectors()
+    {
+        LOCKGUARD(m_dataInspectorGuard);
+        m_dataInspectors.clear();
     }
 
     std::shared_ptr<IDataInspector> LogManagerImpl::GetDataInspector() noexcept
     {
-        return m_dataInspector;
+        if(m_dataInspectors.empty())
+        {
+            return nullptr;
+        }
+        return m_dataInspectors.at(0);
+    }
+
+    std::shared_ptr<IDataInspector> LogManagerImpl::GetDataInspector(const std::string& uniqueIdentifier) noexcept
+    {
+        for (const auto& dataInspector : m_dataInspectors)
+        {
+            std::string inspectorIdentifier(typeid(*dataInspector).name());
```

Comment:
I'm a bit curious about this approach. Instead of iterating over all inspectors, why can't we do something like:
```cpp
auto identifiedInspector = std::find_if(m_dataInspectors.begin(), m_dataInspectors.end(), [&uniqueIdentifier]
{
   return strcmp(inspector->GetName(), uniqueIdentifier) == 0;
});

return identifiedInspector != m_dataInspectors.end() ? identifiedInspector  : nullptr;
```

Reply:
updated